### ISSUE TITLE
Advanced functions

### ISSUE THREAD

**Issue by shreyasminocha** [apesic/rpncalc#7]
Hey. It'd be cool if we had trigonometric functions, a square root key, logarithms, and maybe even things like permutations and combinations. I suppose the UI could take inspiration from the android calculator.

**Comment by stevenbell** [comment]
Looks like it's on the roadmap (see the bottom of the README).  Trig functions be at the top of my list since I use them while woodworking or doing other projects.  For pretty much any other case, I'm already in front of my computer with quick access to Python or similarly powerful tool.

Also wanted to say thanks for making this app!  I especially love the simplicity of the current UI; it seems that most of the RPN calculators try to emulate the look of the venerable HP calculators.  I can see the appeal, but a smartphone is a totally different device and deserves a UI made for touchscreens.  This hits the mark.

**Comment by apesic** [comment]
Yep, the plan is to add an advanced mode similar to the stock Android calculator, with trig functions, logarithms, and a few other of the more commonly used scientific calculator features (buttons for Pi and e, factorials, not sure what else yet). This will be a fairly substantial amount of UI work, so it might be a little while before I get to it.

>Also wanted to say thanks for making this app! I especially love the simplicity of the current UI; it seems that most of the RPN calculators try to emulate the look of the venerable HP calculators. I can see the appeal, but a smartphone is a totally different device and deserves a UI made for touchscreens. This hits the mark.

I'm glad you're enjoying it! What you described is literally the exact reason I made this--as much as I love the old HP calculators, smartphones are a very different form factor and the ideal calculator UI should be designed with that in mind. I kept looking for such an app, and never found that was quite what I wanted so eventually just decided to build it.

**Comment by BenGood** [comment]
I am really enjoying this calculator. Mostly I just want to give some positive feedback. I don't think F-droid has a place to post reviews. 
I would love to have trig and square root functions though. I wish I understood flutter and could help make it happen...

**Comment by foxdan** [comment]
Also would like to chime in with my appreciation of this app, it's fantastic to have a clean smartphone friendly UI for RPN on the go.

There's one function I'd love to see added - Reciprocal `1/x`

Force of habit when doing quick and dirty division calculations entering data first not caring for the order before divide operation then hit `1/x` to correct if it was wrong.

The UI at present is brilliantly clean and functional though, so I wouldn't be sure where to add the button, maybe a long press on the exponent button?

Thanks again for making this app, I couldn't find a "tip jar" address either if that's something you were consider adding to the README (:
